sw/device/tests/aon_timer_wakeup_irq_test.c

```diff
@@ -0,0 +1,162 @@
+
+  // Setup the timer and wait for the IRQ
+  uint32_t sleep_cycles = (sleep_time_us * kClockFreqPeripheralHz / 1000000);
+  LOG_INFO("Setting wakeup interrupt for %u us (%u cycles)", sleep_time_us,
+           sleep_cycles);
+  // Capture the current tick to measure the time the IRQ will take
+  time_elapsed = tick_count_get();
+  aon_timer_testutils_wakeup_config(aon_timer, sleep_cycles);
+  do {
+    wait_for_interrupt();
+  } while (peripheral != kTopEarlgreyPlicPeripheralAonTimerAon &&
+           time_elapsed < sleep_range_h);
+
+  CHECK(time_elapsed < sleep_range_h && time_elapsed > sleep_range_l,
+        "Timer took %u usec which is not in the range %u usec and %u usec",
+        (uint32_t)time_elapsed, sleep_range_l, sleep_range_h);
+
+  CHECK(peripheral == kTopEarlgreyPlicPeripheralAonTimerAon,
+        "Interrupt from incorrect peripheral: exp = %d, obs = %d",
+        kTopEarlgreyPlicPeripheralAonTimerAon, peripheral);
+
+  CHECK(irq == kDifAonTimerIrqWkupTimerExpired,
+        "Interrupt type incorrect: exp = %d, obs = %d",
+        kDifAonTimerIrqWkupTimerExpired, irq);
+}
+
+/** External interrupt handler
+ */
+void ottf_external_isr(void) {
+  // Calc the elapsed time since the activation of the IRQ
+  time_elapsed = tick_count_get() - time_elapsed;
+
+  dif_rv_plic_irq_id_t irq_id;
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&plic, kPlicTarget, &irq_id));
+
+  irq = (dif_aon_timer_irq_t)(
+      irq_id -
+      (dif_rv_plic_irq_id_t)kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired);
+
+  peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
+
+  CHECK_DIF_OK(dif_aon_timer_irq_acknowledge(&aon_timer, irq));
+  // Complete the IRQ by writing the IRQ source to the Ibex specific CC
+  // register.
+  CHECK_DIF_OK(dif_rv_plic_irq_complete(&plic, kPlicTarget, irq_id));
+}
+
+bool test_main(void) {
+  // Enable global and external IRQ at Ibex.
+  irq_global_ctrl(true);
+  irq_external_ctrl(true);
+
+  // Initialize the rv timer to compute the tick
+  tick_init();
+
+  // Initialize pwrmgr
+  dif_pwrmgr_t pwrmgr;
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+
+  // Initialize aon timer
+  CHECK_DIF_OK(dif_aon_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR), &aon_timer));
+
+  dif_pwrmgr_irq_state_snapshot_t pwrmgr_irqs =
+      (dif_pwrmgr_irq_state_snapshot_t)UINT_MAX;
+
+  CHECK_DIF_OK(dif_pwrmgr_irq_restore_all(&pwrmgr, &pwrmgr_irqs));
+
+  // Initialize the PLIC.
+  mmio_region_t plic_base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_plic_init(plic_base_addr, &plic));
+
+  // Enable interrupts
+  rv_plic_testutils_irq_range_enable(
+      &plic, kPlicTarget, kTopEarlgreyPlicIrqIdPwrmgrAonWakeup,
+      kTopEarlgreyPlicIrqIdAonTimerAonWdogTimerBark);
+
+  // Executing the test more than once with different times to make sure the aon
```

Comment:
Rather than call this twice, pick a pseudo-random value by invoking `rand_testutils_gen32_range(1, 50) * 1000`, to test between 1 and 50ms. The randomization should be simulation-seed reproducible.